### PR TITLE
Make kubernetes-version optional and improve launch template validation for upgrade nodegroup

### PR DESCRIPTION
**Pull request by aclevername**
### Description
Improves the UX for this https://github.com/weaveworks/eksctl/issues/3102

Fixes this regression https://github.com/weaveworks/eksctl/issues/3092
- Previously it was always optional, see: https://github.com/weaveworks/eksctl/blob/6cd4d4270c344f6eb1aa2c2c37397f5d7e67d203/pkg/managed/service.go#L158-L162
### Checklist
- [ ] Added tests that cover your change (if possible)
- [ ] Added/modified documentation as required (such as the `README.md`, or the `userdocs` directory)
- [ ] Manually tested
- [ ] Made sure the title of the PR is a good description that can go into the release notes
- [ ] (Core team) Added labels for change area (e.g. `area/nodegroup`), target version (e.g. `version/0.12.0`) and kind (e.g. `kind/improvement`)

### BONUS POINTS checklist: complete for good vibes and maybe prizes?! :exploding_head:
- [ ] Backfilled missing tests for code in same general area :tada:
- [ ] Refactored something and made the world a better place :star2:

### CODE REVIEW

File: pkg/actions/nodegroup/upgrade.go
Comment (cPu1) [inline pkg/actions/nodegroup/upgrade.go:76]: Very minor nit (just something I prefer) that can be ignored:

```suggestion
		if lt == nil || lt.Id == nil || lt.Name == nil {
			return errors.New("cannot update launch template version because the nodegroup is not configured to use one")
		}
		input.LaunchTemplate = &eks.LaunchTemplateSpecification{
			Version: &launchTemplateVersion,
			Id:      describeNodegroupOutput.Nodegroup.LaunchTemplate.Id,
			Name:    describeNodegroupOutput.Nodegroup.LaunchTemplate.Name,
		}
```

**Review by cPu1 — APPROVED** [review]
I have left a minor comment but otherwise LGTM.

File: pkg/actions/nodegroup/upgrade.go
Comment (aclevername) [reply pkg/actions/nodegroup/upgrade.go:76]: I like! 😄

Comment (aclevername) [reply pkg/actions/nodegroup/upgrade.go:76]: I like the refactor, but I think only one of ID or name need to be set?

Comment (cPu1) [reply pkg/actions/nodegroup/upgrade.go:76]: Yes, that makes sense.

**Review by cPu1 — APPROVED** [review]

File: pkg/actions/nodegroup/upgrade.go
Comment (michaelbeaumont) [inline pkg/actions/nodegroup/upgrade.go:76]: So in this case the version isn't changed? Should/can the behavior be the same as [`create nodegroup`](https://github.com/weaveworks/eksctl/blob/7d7a8150bed6ccb6f58f82f634cbf05c3d479b57/pkg/actions/nodegroup/create.go#L242)?

File: pkg/actions/nodegroup/upgrade.go
Comment (michaelbeaumont) [inline pkg/actions/nodegroup/upgrade.go:70]: Just curious, why do we need to set `Name`?

Comment (aclevername) [reply pkg/actions/nodegroup/upgrade.go:70]: [one of the two need to be set](https://github.com/weaveworks/eksctl/pull/2953#discussion_r551967069), its unclear atm if this issue https://github.com/weaveworks/eksctl/issues/3102 is caused because the template only being referenced by the name in EKS, so I've added this as a precaution.

File: pkg/actions/nodegroup/upgrade.go
Comment (aclevername) [reply pkg/actions/nodegroup/upgrade.go:76]: yes the version doesn't get changed. I just copied this from how it used to be to preserve the UX. Whats the behaviour in `create nodegroup`?

Comment (michaelbeaumont) [reply pkg/actions/nodegroup/upgrade.go:76]: If the behavior before was to not update anything when `--version` is left out then it should stay that way I suppose.

I think the default with `create nodegroup` is to use the control plane version, so here I suppose the version would be updated to the control plane version? Like I imagine I should be able to write `upgrade nodegroup --cluster xx --name xx-ng` and it would pick the right version.

Comment (aclevername) [reply pkg/actions/nodegroup/upgrade.go:76]: Perhaps a good change we could make as part of `v1.0`? wdyt?

**Review by michaelbeaumont — APPROVED** [review]
@aclevername Does this definitely fix https://github.com/weaveworks/eksctl/issues/3102 then or are we not sure?

**Comment by aclevername** [comment]
> @aclevername Does this definitely fix #3102 then or are we not sure?

I believe #3201 is just due to a user error, this PR will definitely improve the UX to prevent it from panicking. It does make me unsure if the `name` is needed, but I don't see any harm in keeping it in.

**Comment by aclevername** [comment]
doing 1 more manual test before merging ⚠️

**Comment by aclevername** [comment]
So it does appears to not like having both... very strange. I'm going to change it to use ID if its set, or name if its set.
```
./eksctl upgrade nodegroup --cluster jk-console --name ng-1 --launch-template-version 2
Error: InvalidParameterException: Either provide launch template ID or launch template name in the request.
{
  RespMetadata: {
    StatusCode: 400,
    RequestID: "fc59f526-90e0-4dc1-94c7-6aa3843f4ee2"
  },
  Message_: "Either provide launch template ID or launch template name in the request."
}
```

**Comment by aclevername** [comment]
passed with changes to only set name or ID

**Review by michaelbeaumont — APPROVED** [review]

**Comment by aclevername** [comment]
changed the PR to merge into `release-0.36` and force pushed over the branch to make it be based off `release-0.36` instead of master.

**Review by michaelbeaumont — APPROVED** [review]
LGTM for `0.36`

**Pull request merged**
